nixos/modules/system/boot/stage-2.nix

```diff
@@ -51,14 +61,21 @@ in
         '';
       };
 
-      readOnlyNixStore = mkOption {
-        type = types.bool;
-        default = true;
+      nixStoreMountOpts = mkOption {
+        type = types.listOf types.nonEmptyStr;
+        default = [
+          "ro"
+          "nodev"
+          "nosuid"
+        ];
+        apply = opts: lib.concatStringsSep " " (map lib.escapeShellArg opts);
```

Comment:
Use of apply is generally discouraged because of its awkward semantics. Can you just do this at the usage site?